src/main/perl/Partition.pm

```diff
@@ -97,6 +96,26 @@ sub partition_compare
 
 =pod
 
+=head2 partition_sort
+
+Return sorted C<NCM::Partition> instances (passed as arguments)
+in order of creation.
+
+=cut
+
+# Due to magic of sort global variables, make a function that sorts
+# instead of trying to get the comparison function working properly
+# outside this package
+# (well, easy way would be to use prototype ($$), but we don't like prototypes)
```

Comment:
Remove ambiguous comment or expand on it with the explanation given in the PR discussion.